### ISSUE TITLE
External link checking results in exception: "Invalid argument(s): Truncated URI"

### ISSUE THREAD

**Issue by chalin** [filiph/linkcheck#30]
For the full log, see https://travis-ci.org/flutter/website/jobs/455734598. Here is an excerpt:

```
pub run linkcheck --external --skip-file ./tool/config/linkcheck-skip-list.txt :4002 (logging to /home/travis/tmp/linkcheck-log.txt)
Crawling...
Unhandled exception:
Invalid argument(s): Truncated URI
#0      _Uri._uriDecode (dart:core/uri.dart:2850:13)
#1      Uri.decodeComponent (dart:core/uri.dart:1115:17)
#2      parseHtml.<anonymous closure> (package:linkcheck/src/parsers/html.dart:82:30)
#3      MappedListIterable.elementAt (dart:_internal/iterable.dart:414:29)
#4      ListIterable.toList (dart:_internal/iterable.dart:219:19)
#5      parseHtml (package:linkcheck/src/parsers/html.dart:83:8)
#6      checkPage (package:linkcheck/src/worker/worker.dart:168:10)
<asynchronous suspension>
#7      worker.<anonymous closure> (package:linkcheck/src/worker/worker.dart:192:29)
<asynchronous suspension>
#8      _RootZone.runUnaryGuarded (dart:async/zone.dart:1314:10)
#9      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#10     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#11     _SyncStreamController._sendData (dart:async/stream_controller.dart:763:19)
#12     _StreamController._add (dart:async/stream_controller.dart:639:7)
#13     _StreamController.add (dart:async/stream_controller.dart:585:5)
#14     _RootZone.runUnaryGuarded (dart:async/zone.dart:1314:10)
#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#17     _SyncStreamController._sendData (dart:async/stream_controller.dart:763:19)
#18     _StreamController._add (dart:async/stream_controller.dart:639:7)
#19     _StreamController.add (dart:async/stream_controller.dart:585:5)
#20     _StreamSinkWrapper.add (dart:async/stream_controller.dart:858:13)
#21     _RootZone.runUnaryGuarded (dart:async/zone.dart:1314:10)
#22     CastStreamSubscription._onData (dart:_internal/async_cast.dart:81:11)
#23     _RootZone.runUnaryGuarded (dart:async/zone.dart:1314:10)
#24     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#25     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#26     _SyncStreamController._sendData (dart:async/stream_controller.dart:763:19)
#27     _StreamController._add (dart:async/stream_controller.dart:639:7)
#28     _StreamController.add (dart:async/stream_controller.dart:585:5)
#29     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
...
```

cc @kwalrath @sfshaza2

**Comment by filiph** [comment]
Should be fixed in 2.0.5.

**Comment by chalin** [comment]
Closed by #31
